agent.py

```diff
@@ -0,0 +1,63 @@
+import typer
+import time
+import logging
+from decimal import Decimal
+
+import prediction_market_agent as pma
+from prediction_market_agent.markets.all_markets import MarketType, get_binary_markets
+from prediction_market_agent.agents.abstract import AbstractAgent
+from prediction_market_agent.agents.all_agents import AgentType, get_agent
+from prediction_market_agent.tools.utils import should_not_happen, check_not_none
+from prediction_market_agent.tools.types import Mana, xDai
+
+
+def main(
+    market_type: MarketType = MarketType.MANIFOLD,
+    agent_type: AgentType = AgentType.ALWAYS_YES,
+    sleep_time: int = 1 * 24 * 60 * 60,
+):
+    """
+    Start the agent as a continuous process. Picks a market and answers it, forever and ever.
+    """
+    agent: AbstractAgent = get_agent(agent_type)
+    keys = pma.utils.get_keys()
+
+    while True:
+        # TODO: Agent needs to keep track of the questions it has answered. It should skip them or re-evaluate.
+        available_markets = get_binary_markets(market_type)
+        logging.info(
+            f"Found {len(available_markets)} markets: {[m.question for m in available_markets]}"
+        )
+
+        # TODO: Pick the market with agent-specific logic, for now just pick the first one.
+        market = available_markets[0]  # agent.pick_market(available_markets)
```

Comment:
Since we're taking `available_markets[0]` here, `market` is an instance of the `Market` subclass, as apposed to `MarketProtocol`. So currently `MarketProtocol` is unused.

More generally though, it's not clear why we need separate `Market` and `MarketProtocol` classes. Why can't we hide all the different functionality of omen and manifold markets behind specialized methods on the `Market` base class?

Reply:
> So currently MarketProtocol is unused.

It is used for type-checking.

> all the different functionality of omen and manifold markets behind specialized methods on the Market base class

How do you mean it here?

Something like

```
class Market(BaseModel):
  def get_question(self) -> str:
    return self.question if isinstance(self, ManifoldMarket) else self.title
```

? If so, then I'd say the Protocol gets rid of all the iffing for us. Results in a bit cleaner code and simpler typings.

Reply:
Hopefully I've answered this in the other comment (https://github.com/evangriffiths/prediction-market-agent/pull/14#discussion_r1465173437)